### ISSUE TITLE
python backend streaming pybind11::error_already_set for perf_analyzer with concurrency > 1

### ISSUE THREAD

**Issue by Slyne** [triton-inference-server/server#2855]
**Description**
Try to perf_analyze a simple sequence model with concurrency > 1  (=1 no issues), get error message:
```
terminate called after throwing an instance of 'pybind11::error_already_set'
  what():  error: unpack_from requires a buffer of at least 3489662884 bytes for unpacking 3489662880 bytes at offset 4 (actual buffer size is 132)

At:
  /opt/tritonserver/backends/python/triton_python_backend_utils.py(117): deserialize_bytes_tensor
```

**Triton Information**
21.04

Are you using the Triton container or did you build it yourself?   Triton container

**To Reproduce**
Paste the model.py
```
import numpy as np
import json

# triton_python_backend_utils is available in every Triton Python model. You
# need to use this module to create inference requests and responses. It also
# contains some utility functions for extracting information from model_config
# and converting Triton input/output types to numpy types.
import triton_python_backend_utils as pb_utils
from multiprocessing.pool import ThreadPool

class Decoder(object):
    def __init__(self, blank):
        self.prev = ''
        self.result = ''
        self.blank_symbol = blank
    
    def decode(self, input, start, ready):
        """
        input: a list of characters/a string
        """
        if start:
            self.prev = ''
            self.result = ''
        
        if ready:
            for li in input.decode("utf-8"):
                if li != self.prev:
                    if li != self.blank_symbol:
                        self.result += li
                self.prev = li
        r = np.array([[self.result]], dtype=np.dtype(object))
        return r

class TritonPythonModel:
    """Your Python model must use the same class name. Every Python model
    that is created must have "TritonPythonModel" as the class name.
    """

    def initialize(self, args):
        """`initialize` is called only once when the model is being loaded.
        Implementing `initialize` function is optional. This function allows
        the model to intialize any state associated with this model.
        Parameters
        ----------
        args : dict
          Both keys and values are strings. The dictionary keys and values are:
          * model_config: A JSON string containing the model configuration
          * model_instance_kind: A string containing model instance kind
          * model_instance_device_id: A string containing model instance device ID
          * model_repository: Model repository path
          * model_version: Model version
          * model_name: Model name
        """

        # You must parse model_config. JSON string is not parsed here
        self.model_config = model_config = json.loads(args['model_config'])

        # get max batch size
        max_batch_size = max(model_config["max_batch_size"], 1)

        # get blank symbol from config
        blank = self.model_config.get("blank_id", '-')

        # initialize decoders
        self.decoders = [Decoder(blank) for i in range(max_batch_size)]

        # Get OUTPUT0 configuration
        output0_config = pb_utils.get_output_config_by_name(
            model_config, "OUTPUT0")
            
        # Convert Triton types to numpy types
        self.output0_dtype = pb_utils.triton_string_to_numpy(
            output0_config['data_type'])

    def batch_decode(self, batch_input, batch_start, batch_ready):
        responses = []
        args = []
        idx = 0
        for i,r,s in zip(batch_input, batch_ready, batch_start):
            args.append([idx, i, r, s])
            idx += 1
        
        with ThreadPool() as p:
            responses = p.map(self.process_single_request, args)

        return responses

    def process_single_request(self, inp):
        decoder_idx, input, ready, start = inp
        response = self.decoders[decoder_idx].decode(input[0], start[0], ready[0])
        output0_dtype = self.output0_dtype
        #print("output dtype:", output0_dtype, type(response))
        #print(response.astype(output0_dtype))
        out_tensor_0 = pb_utils.Tensor("OUTPUT0", response.astype(output0_dtype))
        inference_response = pb_utils.InferenceResponse(
                output_tensors=[out_tensor_0])
        return inference_response

    def execute(self, requests):
        """`execute` MUST be implemented in every Python model. `execute`
        function receives a list of pb_utils.InferenceRequest as the only
        argument. This function is called when an inference request is made
        for this model. Depending on the batching configuration (e.g. Dynamic
        Batching) used, `requests` may contain multiple requests. Every
        Python model, must create one pb_utils.InferenceResponse for every
        pb_utils.InferenceRequest in `requests`. If there is an error, you can
        set the error argument when creating a pb_utils.InferenceResponse
        Parameters
        ----------
        requests : list
          A list of pb_utils.InferenceRequest
        Returns
        -------
        list
          A list of pb_utils.InferenceResponse. The length of this list must
          be the same as `requests`
        """
        #print("START NEW")
        responses = []

        batch_input = []
        batch_ready = []
        batch_start = []
        batch_corrid = []
        # Every Python backend must iterate over everyone of the requests
        # and create a pb_utils.InferenceResponse for each of them.
        for request in requests:
            # Get INPUT0
            in_0 = pb_utils.get_input_tensor_by_name(request, "INPUT0")
            #in_0 ->  <triton_python_backend_utils.Tensor object
            # in_0 -> ndarray['xxx']

            batch_input += in_0.as_numpy().tolist()
            
            in_start = pb_utils.get_input_tensor_by_name(request, "START")
            batch_start += in_start.as_numpy().tolist()
            
            in_ready = pb_utils.get_input_tensor_by_name(request, "READY")
            batch_ready += in_ready.as_numpy().tolist()

            in_corrid = pb_utils.get_input_tensor_by_name(request, "CORRID")
            batch_corrid += in_corrid.as_numpy().tolist()

        #print("corrid", batch_corrid)
        #print("batch input", batch_input)
        responses = self.batch_decode(batch_input, batch_start, batch_ready)
        #print("batch response", responses)
        # You should return a list of pb_utils.InferenceResponse. Length
        # of this list must match the length of `requests` list.
        assert len(requests) == len(responses)
        #print("send responses", responses)
        return responses

    def finalize(self):
        """`finalize` is called only once when the model is being unloaded.
        Implementing `finalize` function is OPTIONAL. This function allows
        the model to perform any necessary clean ups before exit.
        """
        print('Cleaning up...')
```

The config.pbtxt
```
name: "ctc_decode_model"
backend: "python"
max_batch_size: 4
sequence_batching{
    max_sequence_idle_microseconds: 5000000
    direct {
      minimum_slot_utilization: 0.1
      max_queue_delay_microseconds: 100000
    }
    control_input [
        {
            name: "START",
            control [
                {
                    kind: CONTROL_SEQUENCE_START
                    fp32_false_true: [0,1]
                }
            ]
        },
        {
            name: "READY"
            control [
                {
                    kind: CONTROL_SEQUENCE_READY
                    fp32_false_true: [0, 1]
                }
            ]
        },
        {
            name: "CORRID",
            control [
                {
                    kind: CONTROL_SEQUENCE_CORRID
                    data_type: TYPE_UINT64
                }
            ]
        }
    ]
}
input [
  {
    name: "INPUT0"
    data_type: TYPE_STRING
    dims: [1] 
  }
]

output [
  {
    name: "OUTPUT0"
    data_type: TYPE_STRING
    dims: [-1]
  }
]

instance_group [{ kind: KIND_CPU }]
```

The commandline is 
```
perf_client --streaming -i gRPC --concurrency-range 2:2:1 -m ctc_decode_model
```

**Comment by Tabrizian** [comment]
Thanks for the report. We will look into it.

**Comment by Slyne** [comment]
@Tabrizian How's it going ?

**Comment by Tabrizian** [comment]
@Slyne We should be able to get to this week.

**Comment by Tabrizian** [comment]
@Slyne This is fixed and it is going to be included in 21.06 release.

**Comment by ioangatop** [comment]
@Tabrizian I have the same issue with `r21.07`

**Comment by Tabrizian** [comment]
@ioangatop Could you please create a new issue with all the details (model repository, your client code, etc.)? I tried the model in this issue again but this problem didn't exist.
